### PR TITLE
Try again when Copr 400 BAD REQUEST is raised

### PR DESCRIPTION
**Pull request by majamassarini**
Probably it is just a race condition.

### CODE REVIEW

**Comment by xsuchy** [comment]
Instead of incrementing attempts and while loop, can we use https://pypi.org/project/backoff/ (we are using this in Copr)

**Review by lbarcziova — COMMENTED** [review]
thanks Maja!

File: packit_service/worker/helpers/build/copr_build.py [hunk 1]
```diff
@@ -893,108 +893,125 @@ def create_or_update_copr_project(self) -> str:
...
+                    "\n"
+                    "Please retrigger the build, once the issue above is fixed.\n"
+                )
+                self.status_reporter.comment(body=msg)
+                raise ex
+            except PackitCoprProjectException as ex:
+                if (
+                    f"{self.job_project}' already exists. Copr HTTP response is 400 BAD REQUEST."
+                    in str(ex)
+                ):
```
[the remark this answers is not in the document]
Comment (majamassarini) [reply packit_service/worker/helpers/build/copr_build.py:995]: I double checked it but still I have doubt. So if you agree I think we can take just the `already exists. Copr HTTP response is 400 BAD REQUEST.` part.

[the remark this answers is not in the document]
Comment (mfocko) [reply packit_service/worker/helpers/build/copr_build.py:995]: I'd prefer `.status_code`, but it seems like it's buried somewhere…

**Review by mfocko — COMMENTED** [review]
Doesn't it use `requests` under the hood? :eyes:

**Comment by majamassarini** [comment]
> Instead of incrementing attempts and while loop, can we use https://pypi.org/project/backoff/ (we are using this in Copr)?

I agree that the looping is not nice but I don't know if for this use case it is worth having a new dependency in Packit. The condition is not so easily matched, I can't just check the response status,  so the context manager could be a bit harder to be used/read. 
And probably we can have the max attempts set to 1 so there should be no need for backoff time, but that is just my guess.

However if the others agree on having a new dependency, I can try it out.

**Comment by xsuchy** [comment]
> but I don't know if for this use case it is worth having a new dependency in Packit. 

It is packaged. If there is some issue in the future, we will have to solve it for Copr anyway. So it will mean no additional workload.

**Comment by majamassarini** [comment]
> Doesn't it use `requests` under the hood? 👀

Good point! I see the copr client is configured to have one retry by default. However the retried errors are 413, 429, 503 (since no one is overriding them). I don't think it is a good idea to retry always when we got a 400 error or a 500 error...
So I would avoid to customize the `requests` retries.

I will try to use the context manager suggested by Mirek instead.

**Comment by majamassarini** [comment]
I close this in favor of https://github.com/packit/packit/pull/2441

**Pull request closed without merging**